src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -53,16 +53,23 @@ public static List<Error> Build(Context context, Document file)
             {
                 if (!context.Config.DryRun)
                 {
-                    if (context.Config.Legacy)
+                    if (context.Config.Legacy || context.Config.StaticOutput)
                     {
                         var output = context.TemplateEngine.RunJavaScript("toc.json.js", JsonUtility.ToJObject(model));
-                        context.Output.WriteJson(outputPath, output);
+                        context.Output.WriteJson(LegacyUtility.ChangeExtension(outputPath, ".json"), output);
```

Comment:
what is this change for?

Reply:
the extension of toc outputPath can be `.html` when the outputType is `Html`

---
In reply to: [411892525](https://github.com/dotnet/docfx/pull/5806#discussion_r411892525) [](ancestors = 411892525)